datalad/interface/common_opts.py

```diff
@@ -33,6 +33,13 @@
     constraints=EnsureInt() | EnsureNone(),
     doc="""limit recursion into subdataset to the given number of levels""")
 
+shared_access_opt = Parameter(
+    args=('--shared-access',),
```

Comment:
I think that would be short-sighted, given the optimizations that are yet to come. Shared access (here), shared annex (we talked about). I cannot foresee if they would be incremental, or orthogonal. Hence I went for a special long option for a special case.